app/views/accounts/new.html.erb

```diff
@@ -65,11 +65,19 @@
         <%= lucide_icon('arrow-left', class: 'text-gray-500 w-5 h-5') %>
       <% end %>
       <span>
-          <% model_class = "Account::#{params[:type]}".constantize %>
-          Add <%= model_class.model_name.human(count: 2) %>
+        <% if params[:type].present? %>
+          <% allowable_types = ["Depository", "Investment", "Property", "Vehicle", "Credit", "Loan", "OtherAsset", "OtherLiability"] %>
+          <% if allowable_types.include?(params[:type]) %>
+            <% model_class = "Account::#{params[:type]}".constantize %>
+            Add <%= model_class.model_name.human(count: 2) %>
+          <% else %>
+            Add Account
+          <% end %>
+        <% else %>
+          Add Account
+        <% end %>
```

Comment:
Super heavy on the if/else statements, which feels a little hard to follow/maintain.

What about something like this? Not 100% positive it'll pass the ruby checks. 
```
<span>
        <% model_class = if params[:type].in?(["Depository", "Investment", "Property", "Vehicle", "Credit", "Loan", "OtherAsset", "OtherLiability"])
                  "Account::#{params[:type]}".constantize
                end %>
        Add <%= model_class ? model_class.model_name.human(count: 2).downcase : "account" %>
      </span>
```

Reply:
Yeah definitely worth a shot, I'll add it now

Reply:
We should centralize/codify the allowable types as I'm sure they will be queried everywhere

```ruby
  # in Account model
  def self.allowable_accountable_types
    accountable_types.map { |type| type.demodulize }
  end
```

Then can do 

```ruby
<% if params[:type].in? Account.allowable_account_types %>
 ... 
<% else %>
   Add Account
<% end %>
```

And probably best to move to a helper to clean up conditionals out of the view.

Reply:
Could it be worthwhile adding new models for singular and plural cases?
```yml
models:
      account/depository:
        singular: Bank Account
        plural: Bank Accounts
```

Reply:
Rails handles that with the `pluralize` methods, so I'm not sure we do (yet)

Reply:
The current implementation has Bank Accounts and Investments as plural by default, should they be changed to singular then?

Reply:
My bad 😆